syntax = "proto3";

import "distribution.proto";

package bayesmix;

message DPPrior {
  message FixedValue {
    double totalmass = 1;
  }

  message GammaPrior {
    GammaDistribution totalmass_prior = 1;
  }

  oneof totalmass {
    FixedValue fixed_value = 1;
    GammaPrior gamma_prior = 2;
  }
}

message PYPrior {
  message FixedValues {
    double strength = 1;
    double discount = 2;
  }

  oneof totalmass {
    FixedValues fixed_values = 1;
  }
}

message LogSBPrior {
  oneof coeff {
    MultiNormalDistribution normal_prior = 1;
  }
  double step_size = 2;  // TODO move?
  uint32 num_components = 3;
}

message TruncSBPrior {
  message BetaPriors {
    repeated BetaDistribution beta_distributions = 1;
  }
  message DPPrior {
    double totalmass = 1;
  }
  message PYPrior {
    double strength = 1;
    double discount = 2;
  }

  oneof prior {
    BetaPriors beta_priors = 1;
    DPPrior dp_prior = 2;
    PYPrior py_prior = 3;
  }
  uint32 num_components = 4;
}
